config/experimental/s3configsource/factory.go

```diff
@@ -0,0 +1,55 @@
+package s3configsource
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/experimental/configsource"
+
+	splunkprovider "github.com/signalfx/splunk-otel-collector"
+)
+
+const (
+	typeStr = "s3"
+)
+
+type (
+	errMissingRegion struct{ error }
+	errMissingBucket struct{ error }
+	errMissingKey    struct{ error }
+)
+
+type s3Factory struct{}
+
+func (s *s3Factory) Type() config.Type {
+	return typeStr
+}
+
+func (s *s3Factory) CreateDefaultConfig() splunkprovider.ConfigSettings {
+	return &Config{
+		Settings: splunkprovider.NewSettings(typeStr),
+	}
+}
+
+func (s *s3Factory) CreateConfigSource(_ context.Context, params splunkprovider.CreateParams, cfg splunkprovider.ConfigSettings) (configsource.ConfigSource, error) {
+	s3Config := cfg.(*Config)
+
+	if s3Config.Region == "" {
```

Comment:
We should be able to infer the default region if not set. We can allow an empty region here.

Reply:
Is there a way to infer region if we don't know/ask for the bucket owner?